src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/LogRecordExtensions.cs

```diff
@@ -145,18 +146,40 @@ internal static OtlpLogs.LogRecord ToOtlpLog(this LogRecord logRecord, SdkLimitO
                     otlpLogRecord.Flags = (uint)logRecord.TraceFlags;
                 }
 
-                int scopeDepth = -1;
                 logRecord.ForEachScope(ProcessScope, otlpLogRecord);
 
                 void ProcessScope(LogRecordScope scope, OtlpLogs.LogRecord otlpLog)
                 {
-                    scopeDepth++;
                     foreach (var scopeItem in scope)
                     {
-                        var scopeItemWithDepthInfo = new KeyValuePair<string, object>($"[Scope.{scopeDepth}]:{scopeItem.Key}", scopeItem.Value);
-                        if (OtlpKeyValueTransformer.Instance.TryTransformTag(scopeItemWithDepthInfo, out var result, attributeValueLengthLimit))
+                        if (scopeItem.Key.Equals("{OriginalFormat}") || string.IsNullOrEmpty(scopeItem.Key))
                         {
-                            otlpLog.AddAttribute(result, attributeCountLimit);
+                            // Ignore if the scope key is empty.
+                            // Ignore if the scope key is {OriginalFormat}
+                            // Attributes should not contain duplicates,
+                            // and it is expensive to de-dup, so this
+                            // exporter is going to pass the scope items as is.
+                            // {OriginalFormat} is going to be the key
+                            // if one uses formatted string for scopes
+                            // and if there are nested scopes, this is
+                            // guaranteed to create duplicate keys.
+                            // Similar for empty keys, which is what the
+                            // key is going to be if user simply
+                            // passes a string as scope.
+                            // To summarize this exporter only allows
+                            // IReadOnlyList<KeyValuePair<string, object?>>
+                            // or IEnumerable<KeyValuePair<string, object?>>.
+                            // and expect users to provide unique keys.
+                            // Note: It is possible that we allow users
+                            // to override this exporter feature. So not blocking
+                            // empty/{OriginalFormat} in the SDK itself.
```

Comment:
Are you suggesting that in the future we might add an exporter option to override this behavior?

My preference would be to try and avoid this solution if it becomes a need in the future. Given we have ILogger specific notions into our LogRecord data model we're required to handle them somehow in our exporters. But, I'd prefer to avoid spilling ILogger specific notions into our OTLP exporter options.

Reply:
deduping is not going to be Ilogger specific. It'll be same for any source which itself prevent duplicates.
For Activity, we do not produce duplicates, so if we want to have that similar behavior, we need to expose something.
(but it could be in the SDK or Exporter...)